File: src/main/java/aurocosh/divinefavor/common/network/message/client/syncing/MessageSyncFireImmunity.java
package aurocosh.divinefavor.common.network.message.client.syncing;

import aurocosh.divinefavor.DivineFavor;
import aurocosh.divinefavor.common.network.base.NetworkWrappedClientMessage;
import net.minecraft.entity.Entity;
import net.minecraft.entity.player.EntityPlayer;
import net.minecraftforge.fml.relauncher.ReflectionHelper;
import net.minecraftforge.fml.relauncher.Side;
import net.minecraftforge.fml.relauncher.SideOnly;

public class MessageSyncFireImmunity extends NetworkWrappedClientMessage {
    public boolean immuneToFire;

    public MessageSyncFireImmunity() {
    }

    public MessageSyncFireImmunity(boolean immuneToFire) {
        this.immuneToFire = immuneToFire;
    }

    @Override
    @SideOnly(Side.CLIENT)
    protected void handleSafe() {
        EntityPlayer player = DivineFavor.proxy.getClientPlayer();
        ReflectionHelper.setPrivateValue(Entity.class,player,immuneToFire,54);
    }
}

File: src/main/kotlin/aurocosh/divinefavor/common/network/message/client/syncing/MessageSyncFireImmunity.kt
package aurocosh.divinefavor.common.network.message.client.syncing

import aurocosh.divinefavor.DivineFavor
import aurocosh.divinefavor.common.network.base.WrappedClientMessage
import net.minecraft.entity.Entity
import net.minecraftforge.fml.relauncher.ReflectionHelper
import net.minecraftforge.fml.relauncher.Side
import net.minecraftforge.fml.relauncher.SideOnly

class MessageSyncFireImmunity : WrappedClientMessage {
    var immuneToFire: Boolean = false

    constructor() {}

    constructor(immuneToFire: Boolean) {
        this.immuneToFire = immuneToFire
    }

    @SideOnly(Side.CLIENT)
    override fun handleSafe() {
        val player = DivineFavor.proxy.clientPlayer
        ReflectionHelper.setPrivateValue(Entity::class.java, player, immuneToFire, 54)
    }
}